api/src/main/java/com/okta/sdk/resource/Resource.java

```diff
@@ -16,10 +16,24 @@
 package com.okta.sdk.resource;
 
 /**
+ * Base representation of a REST resource payload.  All objects transferred over the wire by this SDK are Resources.
  * @since 0.5.0
  */
 public interface Resource {
 
     String getResourceHref();
     void setResourceHref(String href);
+
+    /**
+     * Returns the class this resource represents. This is used when you need to figure out what the main type of
+     * Resource this object is.  For example, DefaultFoo, ImplFoo, and SuperFoo may all represent the Resource of
```

Comment:
Wrap your damn class names with <code> will ya? ;)

Reply:
backticks get you every time on GitHub

Reply:
I really don't like this method.  I used it as a bit of a crutch, and really only for caching.
Each type of resource gets lumped together in a cache (one for User, Group, etc)

Some resources like specific types of Applications inherit from 2 levels of Application parents, but I need to lump them all in the same `Application` bucket.

This is used for cache invalidation, deleting an object removes it's cache. Calling non CRUD PUT/POST methods on an object also invalidates its cache.

This _could_ also be resolved at runtime, but it would require walking the class hierarchy. In the current case, this method is overridden in the generated resource classes:

```java
@javax.annotation.Generated(value = "com.okta.swagger.codegen.OktaJavaClientApiCodegen", date = "2018-01-19T09:23:49.828-05:00")
public interface OpenIdConnectApplication extends Resource, Application {

    default Class<? extends Resource> getResourceClass() {
        return Application.class;
    }
...
```

I'm open for other ideas/thoughts!